spec/other/compatibility_guard_spec.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+RSpec.describe "bundler compatibility guard" do
+  context "when the bundler version is 2+" do
+    before { simulate_bundler_version "2.0.a" }
```

Comment:
Do you foresee the need of testing against various version schemes like Rails started off with SemVer and now it's 5.0.1.1 or 5.1.0.rc1 or 5.1.0.alpha and whatever other extensions they started adding on because they didn't feel like incrementing the version.